crates/wasm-mutate/src/module/mod.rs

```diff
@@ -1,23 +1,62 @@
+use std::convert::TryFrom;
 
+use wasmparser::{Type, TypeDef};
 
-#[derive(Debug)]
+#[derive(Debug, Clone)]
 pub enum PrimitiveTypeInfo {
     I32,
     I64,
     F32,
     F64,
     // TODO, add others
 }
-#[derive(Debug)]
+#[derive(Debug, Clone)]
 pub struct FuncInfo {
     pub params: Vec<PrimitiveTypeInfo>,
     pub returns: Vec<PrimitiveTypeInfo>,
 }
 
-#[derive(Debug)]
+#[derive(Debug, Clone)]
 pub enum TypeInfo {
     Func(FuncInfo),
     Instance(),
-    Module()
+    Module(),
 }
 
+impl TryFrom<Type> for PrimitiveTypeInfo {
+    type Error = super::Error;
+
+    fn try_from(value: Type) -> Result<Self, Self::Error> {
+        match value {
+            wasmparser::Type::I32 => Ok(PrimitiveTypeInfo::I32),
+            wasmparser::Type::I64 => Ok(PrimitiveTypeInfo::I64),
+            wasmparser::Type::F32 => Ok(PrimitiveTypeInfo::F32),
+            wasmparser::Type::F64 => Ok(PrimitiveTypeInfo::F64),
+            _ => Err(super::Error::InvalidTypeMapping)
```

Comment:
Nitpick: I'd call this `UnsupportedType` or something like that and then carray along the `wasmparser::Type` as a payload within the enum variant to make better error messages.

Reply:
Addressed in PR 337